### PR TITLE
Added `CERTIFICATE_VERIFY_MAX_LENGTH` constant

### PR DESCRIPTION
**Pull request by VeronikaNguyen**
- Set `CERTIFICATE_VERIFY_MAX_LENGTH` to `65539 = 2^16 - 1 + 2 + 2`
- Changed `SSL3_RT_MAX_PLAIN_LENGTH` to `CERTIFICATE_VERIFY_MAX_LENGTH`  in `case TLS_ST_CR_CERT_VRFY` in the function `ossl_statem_client_max_message_size` of `statem_clnt.c` and in the function `ossl_statem_server_max_message_size` of `statem_srvr.c`
- Addresses [#20478](https://github.com/openssl/openssl/issues/20478)

### CODE REVIEW

**Review by baentsch — CHANGES_REQUESTED** [review]
@VeronikaNguyen You are aware that you(r employer) needs to provide the [CLA](https://www.openssl.org/policies/cla.html) before this can move forward? Second comment: For testing these changes, may it be sensible to extend the artificial "signature algorithm" in the [test tls-provider](https://github.com/openssl/openssl/blob/master/test/tls-provider.c) to a(n even more artificial) length that actually triggers/necessitates these constant changes? It might be as simple as setting a suitably large number here: https://github.com/openssl/openssl/blob/931369429564b5a9bb09711de8e885fef546a0ac/test/tls-provider.c#L2747 and here https://github.com/openssl/openssl/blob/931369429564b5a9bb09711de8e885fef546a0ac/test/tls-provider.c#L2766  (maybe dependent on the changed constants?). That way we can be sure it's not just `oqsprovider` that tests these limits.

**Comment by VeronikaNguyen** [comment]
Sorry for the late answer. Thanks @baentsch for the review. Yes, I submitted a CLA on Friday. I will look into your testing proposal and update the PR this week.

File: ssl/statem/statem_local.h [hunk 1]
```diff
@@ -31,6 +31,9 @@
 /* Max ServerHello size permitted by RFC 8446 */
 #define SERVER_HELLO_MAX_LENGTH         65607
 
+/* Max CertificateVerify size permitted by RFC 8446 */
+#define CERTIFICATE_VERIFY_MAX_LENGTH   65538
```
Comment (mattcaswell) [inline ssl/statem/statem_local.h:35]: I calculate this as 65539...am I wrong?

````
      struct {
          SignatureScheme algorithm;
          opaque signature<0..2^16-1>;
      } CertificateVerify;
````

`2^16 - 1` == 65535
plus 2 "length" bytes for the signature
plus 2 byes to identify the signature scheme
== 65539

Comment (VeronikaNguyen) [reply ssl/statem/statem_local.h:35]: Good catch! I will update it.

**Comment by mattcaswell** [comment]
We don't accept merge commits. Can you remove the last merge commit? Thanks!

**Comment by mattcaswell** [comment]
Aside from that - this PR looks good to me.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
It looks like the author on the commits is not the same as on your submitted CLA.  Could you change the author via `git commit --amend --author=""` please.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Squashed the commits and merged to master branch. Thank you for your contribution.

**Pull request closed without merging**